src/modules/cmdpal/Microsoft.CmdPal.UI/LoadIconBehavior.xaml.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CmdPal.Extensions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.Xaml.Interactivity;
+
+namespace Microsoft.CmdPal.UI;
+
+public partial class LoadIconBehavior : DependencyObject, IBehavior
+{
+    public IconDataType Source
+    {
+        get => (IconDataType)GetValue(SourceProperty);
+        set
+        {
+            SetValue(SourceProperty, value);
+            OnSourcePropertyChanged();
+        }
+    }
+
+    // Using a DependencyProperty as the backing store for Source.  This enables animation, styling, binding, etc...
+    public static readonly DependencyProperty SourceProperty =
+        DependencyProperty.Register("Source", typeof(IconDataType), typeof(LoadIconBehavior), new PropertyMetadata(new IconDataType(string.Empty)));
+
+    public DependencyObject? AssociatedObject { get; private set; }
+
+    public void Attach(DependencyObject associatedObject) => AssociatedObject = associatedObject;
+
+    public void Detach() => AssociatedObject = null;
+
+    public async void OnSourcePropertyChanged()
+    {
+        var iconService = App.Current.Services.GetService<IIconCacheService>()!;
+
+        var icoSource = await iconService.GetIconSource(Source ?? new(string.Empty));
+
+        if (AssociatedObject is Border border)
+        {
+            if (icoSource is FontIconSource fontIco)
+            {
+                fontIco.FontSize = border.Width;
+
+                // For inexplicable reasons, FontIconSource.CreateIconElement
+                // doesn't work, so do it ourselves
```

Comment:
Well, that's surprising! Is there an open issue for this in the WinUI repo?

I know there was some weird discussion around resourcing and the XAML tree recently, but think that was for `Geometry`s, but I wonder if there'd be some other odd shared reference with the font info/glyph maybe, who knows.

Reply:
Nothing that I found. I'm not sure if it's a platform bug or a "I'm holding it wrong" thing. IIRC it did end up returning an icon, but it never made it into the UI tree? 🤷